pkg/scyllaclient/client_rclone_agent_integration_test.go

```diff
@@ -44,6 +46,134 @@ func TestRcloneS3ListDirAgentIntegration(t *testing.T) {
 	}
 }
 
+func TestRcloneDeletePathsInBatchesAgentIntegration(t *testing.T) {
+	var (
+		ctx      = context.Background()
+		testHost = ManagedClusterHost()
+		dirName  = "tmp/copy"
+	)
+
+	S3InitBucket(t, testBucket)
+	client, err := scyllaclient.NewClient(scyllaclient.TestConfig(ManagedClusterHosts(), AgentAuthToken()), log.NewDevelopmentWithLevel(zapcore.ErrorLevel))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	const bigSize = 3456
+	lotsOfFiles := make([]string, 0, bigSize)
+	for i := 0; i < bigSize; i++ {
+		lotsOfFiles = append(lotsOfFiles, fmt.Sprint(i))
+	}
+
+	testCases := []struct {
+		name      string
+		before    []string
+		delete    []string
+		after     []string
+		batchSize int
+	}{
+		{
+			name:      "delete all",
+			before:    []string{"a", "b", "c"},
+			delete:    []string{"a", "b", "c"},
+			after:     nil,
+			batchSize: 1,
+		},
+		{
+			name:      "delete only non-existing files",
+			before:    []string{"a", "b", "c", "d"},
+			delete:    []string{"e", "f", "g"},
+			after:     []string{"a", "b", "c", "d"},
+			batchSize: 5,
+		},
+		{
+			name:      "delete subset with non-existing file",
+			before:    []string{"a", "b", "c", "d"},
+			delete:    []string{"a", "c", "e"},
+			after:     []string{"b", "d"},
+			batchSize: 2,
+		},
+		{
+			name:      "delete thousands of files",
+			before:    append([]string{"a"}, lotsOfFiles...),
+			delete:    append([]string{}, lotsOfFiles...),
+			after:     []string{"a"},
+			batchSize: 1000,
+		},
```

Comment:
Thanks ! Looks good,
Please add 'delete nothing' edge case.
```
		{
			name:      "delete nothing",
			before:    []string{"a", "b", "c"},
			delete:    []string{},
			after:     []string{"a", "b", "c"},
			batchSize: 1,
		},
```
It PASSES according to my check done locally.

Reply:
It's actually partially by mistake.
`RcloneDeletePaths` returns error on empty delete (because rclone implementation that I wrote returns an error for safety), but `RcloneDeletePathsInBatches` does not (it doesn't send any request to rclone server).
So this is somewhat of a gray area, but I can add this case to document this behavior.